microscope/batch/batch.py

```diff
@@ -1,17 +1,31 @@
 import time
-import queue as queuemodule
 import sys
 
+import queue as queuemodule
+
 from microscope.monitor.monitor import MonitorRunner
 
 
 def batch(runner: MonitorRunner, timeout: int):
     start_time = time.time()
     while(runner.is_alive() and runner.close_queue.empty()
           and (start_time + timeout > time.time() or timeout == 0)):
+        while True:
```

Comment:
Would the repeated while-true loop be better factored out? then the inner behaviour will have a name (the function).

Reply:
Done.